kubemarine/patches/strong_ciphers_patch.py

```diff
@@ -0,0 +1,113 @@
+        if probable_ciphers_initial == desired_ciphers:
+            cluster.log.info("API server already has the desired cipher suites in the initial inventory. Skipping patch.")
+            return
+
+        kubernetes_nodes = cluster.make_group_from_roles(['control-plane'])
+
+        for member_node in kubernetes_nodes.get_ordered_members_list():
+            apiserver_file = "/etc/kubernetes/manifests/kube-apiserver.yaml"
+
+            # Load the YAML configuration
+            try:
+                apiserver_config = yaml.safe_load(member_node.sudo(f"cat {apiserver_file}").get_simple_out())
+            except yaml.YAMLError as exc:
+                cluster.log.error(f"Failed to parse YAML file: {exc}")
+                continue
+
+            # Check for any existing --tls-cipher-suites argument
+            existing_ciphers_found = False
+            for command in apiserver_config['spec']['containers'][0]['command']:
+                if command.startswith("--tls-cipher-suites="):
+                    existing_ciphers_found = True
+                    break
+
+            # Compare with desired ciphers
+            if existing_ciphers_found:
+                cluster.log.info("Skipping patch on this node as tls-cipher-suites are already present.")
+                continue
+
+            # Apply the patch
+            apiserver_config['spec']['containers'][0]['command'].append("--tls-cipher-suites=TLS_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_CHACHA20_POLY1305_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_RSA_WITH_3DES_EDE_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_256_GCM_SHA384")
+            updated_config = yaml.dump(apiserver_config)
+            member_node.put(StringIO(updated_config), apiserver_file, backup=True, sudo=True)
+
+            # Call verify_apiserver_restart to check API server restart
+            restart_successful = self.verify_apiserver_restart(member_node, res)
+
+            if restart_successful:
+                cluster.log.info("API server restart succeeded.")
+            else:
+                cluster.log.error("API server restart failed.") 
+
+    def verify_apiserver_restart(self, member_node: NodeGroup, res: DynamicResources) -> bool:
```

Comment:
There are a few bugs and gaps below that would not arise if you followed one of existing approaches mentioned in https://github.com/Netcracker/KubeMarine/pull/581#discussion_r1458574834

Reply:
Followed the existing approach as specified in the comment